types/precison_timestamp_types.go

```diff
@@ -0,0 +1,147 @@
+package types
+
+import (
+	"fmt"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+// TimePrecision is used to represent the precision of a timestamp
+type TimePrecision int32
+
+const PrecisionUnknown TimePrecision = -1
```

Comment:
this doesn't need to be a separate block, it can still be grouped in the block below without changing the values

Reply:
I did try this initially, but iota gets reset to zero on first statement of block and subsequently gets incremented for each statement. If I include unknown, it was giving value 1 for seconds ([goPlayground](https://go.dev/play/p/kK4EhOKAUQw))

Reply:
ah interesting, I didn't realize it worked out that way. fair enough